server/routes/creatingReports/incidentDetails.test.ts

```diff
@@ -65,6 +68,42 @@ describe('GET /section/form', () => {
         expect(locationService.getIncidentLocations).toBeCalledWith('user1-system-token', 'current-agency-id')
       })
   })
+  test('should render saved data', () => {
+    draftReportService.getCurrentDraft.mockResolvedValue({
+      form: {
+        incidentDetails: {
+          plannedUseOfForce: true,
+          authorisedBy: 'Eric Bloodaxe',
+        },
+      },
+    })
+    return request(app)
+      .get(`/report/1/incident-details`)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Eric Bloodaxe')
+        expect(locationService.getIncidentLocations).toBeCalledWith('user1-system-token', 'current-agency-id')
+      })
+  })
+
+  test('should preference flash state over persisted data', () => {
```

Comment:
Rather than testing for something that isn't there, would it be more definitive to test for something that is?  
~ setting a value for 'authorisedBy' in flash i.e 
flash.mockReturnValue([ plannedUseOfForce: true, authorisedBy: 'some person' ])
expect(res.text).toContain('some person)

?

Reply:
You are totally right that it's not nice testing something that isn' there.

so this is a bit weird, if any value is on the key in flash it would have worked correctly - it only takes into account when the key is missing from flash.

this is because two objects were splatted on top of each other:
```
const a = { bob: true }
const b = { bob: false}

const c =  { ...a, ...b}  // => { bob: false}
```

This wouldn't work when we removed the `b` key: 
```
const a = { bob: true }
const b = { }

const c =  { ...a, ...b}  // => wanted '{}' got '{ bob: true}'
```
So had to change the implementation to not use splatting

The way I fixed this was write this failing test first and then fixing it to make the test pass.

Reply:
cool